File: src/CQRS.EventHandlers.FSharp.Tests/Ordered.Facts.fs
namespace CQRS.EventHandlers.FSharp.Tests

open System
open Xunit
open CQRS.EventHandlers.FSharp
open CQRS.EventHandlers.Tests

module ``Ordered facts`` = 

    [<Trait (Constants.Category, Constants.FSharp)>]
    type ``Ordered Decorator HandleEvent facts`` () = 
        inherit Ordered_Decorator_HandleEvent_Facts ()

        override this.Decorate handler = 
            Ordered.Decorator.Decorate handler

File: src/CQRS.EventHandlers.CSharp.Tests/Ordered.Facts.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using CQRS.EventHandlers.CSharp.Behaviours.Ordered;
using Moq;
using Xunit;
using Base = CQRS.EventHandlers.Tests;

namespace CQRS.EventHandlers.CSharp.Tests {
    [Trait(Base.Constants.Category, Base.Constants.CSharp)]
    public class Ordered_Decorator_HandleEvent_Facts : Base.Ordered_Decorator_HandleEvent_Facts {
        protected override IEventHandler<Base.FakeEntity> Decorate(IEventHandler<Base.FakeEntity> handler) {
            return new Decorator<Base.FakeEntity>(handler);
        }
    }
}
